app/admin/quests/create/page.tsx

```diff
@@ -0,0 +1,1049 @@
+                      ...prev,
+                      token: event.target.value,
+                      token_decimals:
+                        TOKEN_DECIMAL_MAP[
+                          event.target.value as keyof typeof TOKEN_DECIMAL_MAP
+                        ],
+                    }));
+                  }}
+                  options={Object.keys(TOKEN_ADDRESS_MAP[network]).map(
+                    (eachItem) => {
+                      return {
+                        value: eachItem,
+                        label: eachItem,
+                      };
+                    }
+                  )}
+                />
+                <Textinput
+                  onChange={handleBoostInputChange}
+                  value={boostInput.amount}
+                  name="amount"
+                  label="Amount"
+                  placeholder="Amount"
+                />
+              </div>
+            ) : null}
+          </InputCard>
+
+          <div className="w-full items-center justify-center flex">
+            <div className="w-fit">
+              <Button
+                loading={buttonLoading}
+                onClick={async () => {
+                  const id = await handleCreateQuest();
+                  if (!id) return;
+                  await handleCreateBoost(id);
+                  handlePagination("Next");
+                }}
+                disabled={isButtonDisabled}
+              >
+                <p>Confirm Next</p>
+              </Button>
+            </div>
+          </div>
+          <QuizControls
+            step={currentPage}
+            setStep={setCurrentPage}
+            onCancel={() => router.push("/admin/quests")}
+          />
+        </>
+      ) : currentPage === 3 ? (
+        <div className="flex flex-col gap-8 w-full">
+          <p className={styles.cardHeading}>{headingText}</p>
+          {steps.map((step, index) => (
+            <InputCard key={index}>
+              <div className="flex gap-8">
+                <div className="flex flex-row w-full gap-2 bg-gray-300 p-4 rounded-xl">
+                  <>
+                    {TASK_OPTIONS.map((category) => (
+                      <div
+                        onClick={() => {
+                          if (
+                            category === "Twitter" ||
+                            category === "Twitter"
+                          ) {
+                            setShowTwitterOption(index);
+                            return;
+                          }
+                          setShowTwitterOption(-1);
+                          setSteps((prev) => {
+                            const newArr = [...prev];
+                            newArr[index] = {
+                              type: category as TaskType,
+                              data: getDefaultValues(category as TaskType),
+                            };
+                            return newArr;
+                          });
```

Comment:
In a function. 

This applies to all difficult inline functions you wrote (I won't see most of them I guess).